lib/Db/TagMapper.php

```diff
@@ -145,20 +145,22 @@ public function getAllTagsForMessages(array $messages): array {
 
 		$qb = $this->db->getQueryBuilder();
 		$idsQuery = $qb->select('mt.*')
+			->addSelect('t.*')
 			->from('mail_message_tags', 'mt')
+			->join('mt', $this->getTableName(), 't', $qb->expr()->eq('mt.tag_id','t.id'))
 			->where(
-				$qb->expr()->in('imap_message_id', $qb->createNamedParameter($ids, IQueryBuilder::PARAM_STR_ARRAY))
+				$qb->expr()->in('mt.imap_message_id', $qb->createNamedParameter($ids, IQueryBuilder::PARAM_STR_ARRAY))
 			);
 		$idsQuery = $idsQuery->execute();
 		$queryResult = $idsQuery->fetchAll();
 		if (empty($queryResult)) {
 			return [];
 		}
-		$result = [];
-		foreach ($queryResult as $qr) {
-			$result[] = $qr['imap_message_id'];
-			$result[$qr['imap_message_id']][] = $this->getTag((int)$qr['tag_id']);
-		};
+		$result = array_map(function ($qr) {
+			$transformed[] = $qr['imap_message_id'];
+			$transformed[$qr['imap_message_id']] = $qr;
```

Comment:
and here you write to a specific index. Could you elaborate on that logic?

Reply:
see as above. I realise now that i need to add an additional [] accessor to this otherwise it will only ever show one tag